network/src/noise/stream.rs

```diff
@@ -672,4 +670,47 @@ mod test {
         block_on(server.read_exact(&mut buf_receive)).unwrap();
         assert_eq!(&buf_receive[..], &buf_send[..]);
     }
+
+    #[test]
+    fn fragmented_stream() {
+        // create an in-memory socket for testing
+        let (mut dialer_socket, mut listener_socket) = ExposingTestSocket::new_pair();
+
+        // fragment reads
+        dialer_socket.set_fragmented_read();
+        listener_socket.set_fragmented_read();
+
+        // get peers
+        let ((client, _client_public_key), (server, server_public_key)) = build_peers();
+
+        // perform the handshake
+        let (client, server) = block_on(join(
+            client.upgrade_outbound(dialer_socket, server_public_key, AntiReplayTimestamps::now),
+            server.upgrade_inbound(listener_socket),
+        ));
+
+        // get session
+        let mut client = client.unwrap();
+        let (mut server, _) = server.unwrap();
+
+        //
+        block_on(client.write_all(b"The Name of the Wind")).unwrap();
```

Comment:
Nit: I'd normally like to see these in a variable and reused twice, especially then you don't accidentally type another space or whatever,  but the readability is nice here.